cdap-hbase-spi/src/main/java/co/cask/cdap/spi/hbase/HBaseDDLExecutor.java

```diff
@@ -17,13 +17,29 @@
 package co.cask.cdap.spi.hbase;
 
 
+import java.io.Closeable;
 import java.io.IOException;
 import javax.annotation.Nullable;
 
 /**
  * Interface providing the HBase DDL operations.
  */
-public interface HBaseDDLExecutor {
+public interface HBaseDDLExecutor extends Closeable {
+
+  /**
+   * Initialize the {@link HBaseDDLExecutor} with the configuration object provided
+   * @param conf the configurations with which to initialize
+   * @param <T>
+   */
+  <T> void initialize(T conf);
+
+  /**
+   * Returns the conf with which the {@link HBaseDDLExecutor} was initialized.
+   * @param <T>
+   * @return the conf
+   */
+  <T> T getConf();
```

Comment:
This is required because https://github.com/caskdata/cdap/blob/develop/cdap-hbase-compat-base/src/main/java/co/cask/cdap/data2/util/hbase/HBaseTableUtil.java#L251 here we use conf to get the compression. But by looking at the comment on the method, seems like this is only used for test case though. Will see if can be removed in the next PR when we pass the `TableDescriptor`